src/test/java/seedu/address/model/tut/TutorialTest.java

```diff
@@ -175,16 +176,123 @@ public void setAttendance_studentNotInTut_false() {
+        tutorial.add(ALICE);
+        tutorial.deleteStudent(ALICE);
+        assertFalse(tutorial.studentInList(ALICE));
+    }
+
+    @Test
+    public void studentInListTest() {
+        Tutorial tutorial = Tutorial.of(new TutName(TUT_NAME), TUTORIAL_CLASS);
+        tutorial.add(ALICE);
+        assertTrue(tutorial.studentInList(ALICE));
+    }
+
+    @Test
+    public void deleteStudentTest() {
+        Tutorial tutorial = Tutorial.of(new TutName(TUT_NAME), TUTORIAL_CLASS);
+        tutorial.add(ALICE);
+        assertTrue(tutorial.studentInList(ALICE));
+        tutorial.deleteStudent(ALICE);
+        assertFalse(tutorial.studentInList(ALICE));
+    }
+
+    @Test
+    public void constructor_none_singleton() {
+        Tutorial none = Tutorial.none();
+        Tutorial none1 = Tutorial.none();
+        assertEquals(none, none1);
+    }
+
+    @Test
+    public void addStudent_noneTest() {
+        NONE.add(ALICE);
+        assertTrue(NONE.getStudents().contains(ALICE));
+    }
+
+    @Test
+    public void addStudent_nullStudent_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> NONE.add(null));
+    }
+
+    @Test
+    public void setAttendance_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.setAttendance(new Date(), new StudentId("1000")));
+    }
+
+    @Test
+    public void getTutName_noneTest() {
+        assertThrows(NoTutorialException.class, NONE::getTutName);
+    }
+
+    @Test
+    public void tutorialDateInList_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.tutorialDateInList(TUT_DATE));
+    }
+
+    @Test
+    public void addTutorialDate_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.addTutorialDate(TUT_DATE));
+    }
+
+    @Test
+    public void getTutorialDate_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.getTutorialDate(new Date()));
+    }
+
+    @Test
+    public void isValidTutorialDate_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.isValidTutorialDate(TUT_DATE));
+    }
+
+    @Test
+    public void getTutDates_noneTest() {
+        assertThrows(NoTutorialException.class, NONE::getTutDates);
+    }
+
+    @Test
+    public void getTutorialClass_noneTest() {
+        assertThrows(NoTutorialException.class, NONE::getTutorialClass);
+    }
+
+    @Test
+    public void markAttendance_noneTest() {
+        assertThrows(NoTutorialException.class, ()
+                -> NONE.markAttendance(ALICE, TUT_DATE));
+    }
+
+    @Test
+    public void studentInList_noneTest() {
+        NONE.add(ALICE);
+        assertTrue(NONE.studentInList(ALICE));
+    }
+
+    @Test
+    public void deleteStudent_noneTest() {
+        NONE.add(ALICE);
+        assertTrue(NONE.studentInList(ALICE));
+        NONE.deleteStudent(ALICE);
+        assertFalse(NONE.studentInList(ALICE));
+    }
```

Comment:
Comprehensive testing on the latest changes on implementation!! Everything looks good to me!